### DISCUSSION TITLE
Tanya Error code shiny

### DISCUSSION THREAD

**Discussion by AjiPramasto** [teamalgoritma/community#246] [Q&A] [answered]
Halo,

Mohon bantuannya untuk error di bawah ini:
![image](https://user-images.githubusercontent.com/107389614/182753160-048ab9b4-b660-4d2d-a3dd-31d1fd3b829a.png)

terima kasih

**Comment by fafilia** [comment, marked as answer]
Halo @AjiPramasto , berdasarkan pesan error yang muncul sepertinya disebabkan oleh kesalahan pada penggunaan fungsi lapply. Apakah Bapak menggunakan fungsi tersebut? Mungkin boleh dicopykan code yang ada lapplynya pak?

**Reply by AjiPramasto** [reply]
Oke, makasih Mba, sudah bisa.

**Reply by AjiPramasto** [reply]
Kalau muncul erro ini gmn ya Mba:
![image](https://user-images.githubusercontent.com/107389614/182758784-3e95b0f5-49e4-43d9-80e6-7925c17d635a.png)
mohon bantuannya

**Reply by fafilia** [reply]
Untuk panel sebelah kanan yang muncul pesan error, apakah Bapak ingin menampilkan plot tertentu? atau pesan tersebut muncul setelah running/memilih select input pada bar plot di sebelah kirinya?

**Reply by AjiPramasto** [reply]
iya Mba, kode plotnya tidak jalan, kalau select filter sudah bisa di plot yang var, ini kode plot yg error:

output$Trend_line <- renderPlotly({
    
    data_sales <- data %>%
      group_by(Year, Month, Bulan) %>%
      filter(Cabang == "input_category") %>% 
      summarise(Net_sales = sum(Net_sales),
                Sales_ach = mean(Sales_ach)*100) %>% 
      ungroup()
    
    plot_sales_trend <- data_sales %>% 
      mutate(label = glue("Net Sales: {scales::comma(Net_sales)} Ton
                      Sales Achievement: {Sales_ach} %"))
    
    line_sales_trend <- plot_sales_trend %>% 
      ggplot( aes(x = reorder(Bulan, Month),
                  y = Net_sales, 
                  color = Year, 
                  group = Year,
                  text = label))+
      geom_line()+
      geom_point()+
      labs(title = "Net Sales Trend 2018 vs 2017",
           x = NULL, y = "In Ton",
           color = "Year")+
      theme_minimal()+
      scale_y_continuous(breaks = seq(0,10000,1000),labels = scales::comma)+
      theme(legend.position = "top",
            axis.title.y = element_text(margin = margin(r = 20)), #untuk mengatur posisi sumbu y
            plot.title = element_text(hjust = 0.5))
    
    
    ggplotly(line_sales_trend, tooltip = "text")
    
  
  })  


mohon bantuannya

**Reply by AjiPramasto** [reply]
ini screenshoot errornya:
![image](https://user-images.githubusercontent.com/107389614/182764466-9c2162b5-584e-4008-89d4-2bcaa9b51f4e.png)

**Reply by fafilia** [reply]
Apakah disini Bapak ingin menampilkan lineplot untuk setiap Cabang?
Jika dilihat dari code yang dikirimkan, Bapak ingin menambahkan input pada bagian filter yangmana codenya dapat diubah menjadi:

```
filter(Cabang %in% input$input_category)
```

Selain itu, struktur piping bisa diubah dengan urutan filter, groupby, summarise, ungroup. Silahkan dicoba terlebih dahulu.

**Reply by fafilia** [reply]
Di luar shiny apakah code untuk plotnya sudah dicoba dan berhasil pak? Sebagai saran sebelum diimplementasikan ke shiny boleh dipastikan terlebih dahulu bahwa plot statis sudah berjalan dengan baik terlebih dahulu sehingga tracing codenya lebih mudah untuk mengetahui errornya karena apa.

**Reply by AjiPramasto** [reply]
kalau di rmd reguler, ini kodenya:

```{r}
plot <- Data_grafik_sales %>%
  mutate(label = glue("Total Net Sales: {scales::comma(Total_ach_NetSales)} Ton
                      Sales Achievement: {Sales_ach} %"))

plot1 <- plot %>% ggplot(aes(x = reorder(Bulan, Month),  y = Total_ach_NetSales, color = Year, group = Year, text = label))+
   geom_line()+
   geom_point()+
    labs(title = "Net Sales Trend 2018 vs 2017",
        x = NULL, y = "In Ton",
        color = "Year")+
    theme_minimal()+
  scale_y_continuous(breaks = seq(0,10000,500),labels = scales::comma)+
    theme(legend.position = "bottom",
    axis.title.y = element_text(margin = margin(r = 25)),
    plot.title = element_text(hjust = 0.5))

ggplotly(plot1, tooltip = "text")
```
![image](https://user-images.githubusercontent.com/107389614/182774761-894bc3a7-f893-413b-adc3-40c852c53401.png)

urutan sudah dirubah, dan pakai %in% juga sudah dicoba, masih belum muncul

**Comment by fafilia** [comment]
Oke baik. Mungkin bisa dikonfirmasi terlebih dahulu pak, untuk kasus ini berarti ketika user pilih salah satu Cabang dari selectInput nanti akan muncul 2 plot yang informasinya sama ya pak yaitu dalam bentuk bar plot dan line plot? Apakah seperti itu?

**Reply by AjiPramasto** [reply]
betul mba, yang bar hanya total per tahun sedangkan line adalah per bulan dan tahunnya (sesuai gbr di ats)

**Reply by fafilia** [reply]
Baik, jika casenya seeprti itu berarti untuk id input di kedua plot sama, sedangkan untuk id outputnya saja pak yang berbeda karena hasil outputnya juga berbeda. Boleh dicopykan pak untuk code UI yang khusus untuk menampilkan plot line?

**Reply by AjiPramasto** [reply]
berikut:

 # --------- PLOT
              fluidRow(
                # lollipop (kiri)
                # fungsi box tidak diberikan argumen title karena mengikuti title dari server
                box(width = 4,
                  plotlyOutput(outputId = "Yearly_bar")
                ),
                
                # lineplot (kanan)
                box(width = 8,
                  plotlyOutput(outputId = "Trend_line")
                  )
                  
                )
                
              ),
